flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -1389,6 +1381,109 @@ private void copyStateDataHandleData(
 		return Tuple2.of(stateInfo.f0, newMetaInfo);
 	}
 
+	private <N, S extends State, SV> void migrateStateIfNecessary(
```

Comment:
Could you not still let this method return the meta info, changed or unchanged, just to document the fact that the meta info can be modified by this method? Not a fan of methods that just silently modify their input parameters if it can be avoided.